flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/DeclineCheckpoint.java

```diff
@@ -34,21 +35,18 @@
 	private static final long serialVersionUID = 2094094662279578953L;
 
 	/** The reason why the checkpoint was declined. */
+	@Nullable
 	private final Throwable reason;
```

Comment:
Shouldn't we change the type if we know that it is always serialized?

Reply:
The same actually applies to the return type of `getReason()` method.

Reply:
I guess that we can now also adapt `CheckpointCoordinator.getCheckpointException`.

Reply:
I did the first two things, but didn't know how to proceed with the third: the method is also called from `onTriggerFailure` where we just get a `Throwable` from the future.

Reply:
I overlooked the other usages of this method. I guess it should be fine then.